transaction/std/src/transaction_builder.rs

```diff
@@ -38,6 +38,22 @@ impl TxOutputsOrdering for DefaultTxOutputsOrdering {
     }
 }
 
+/// Transaction output context is produced by add_output method
+/// Used for receipt creation
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct TxOutContext {
+    /// TxOut that comes from a transaction builder
+    /// add_output/add_change_output
+    pub tx_out: TxOut,
+    /// confirmation that comes from a transaction builder
+    /// add_output/add_change_output
+    pub confirmation: TxOutConfirmationNumber,
+    /// Shared Secret that comes from a transaction builder
+    /// add_output/add_change_output
+    pub shared_secret: RistrettoPublic,
```

Comment:
This is a fairly sensitive piece of information, I'm not sure it should be coming out of transaction builder at all. The shared secret is derived via DH key exchange and (I think) can be derived on the other side/via ffi as long as the Moby app has the view private key. 

At the very least it shouldn't be a public field.